e2e/create_test_component.go

```diff
@@ -18,256 +16,256 @@ import (
+		WithStep("", 2, setup.AddComponentVersions(podinfoRedis(t, privateKey, keyName, componentNameIdentifier, testPath, version))).
+		WithStep("", 2, setup.AddComponentVersions(podinfo(t, privateKey, keyName, componentNameIdentifier, testPath, version)))
 }
 
-func podinfo(t *testing.T, componentNameIdentifier string) setup.Component {
+func podinfo(t *testing.T, privateKey []byte, privateKeyName string, componentNameIdentifier string, testPath string, version string) setup.Component {
 	t.Helper()
 
-	temp := setup.Component{
-		Component: shared.Component{
-			Name:    componentNamePrefix + componentNameIdentifier + podinfoComponentName,
-			Version: "1.0.0",
-		},
-		Repository: "podinfo",
+	return setup.Component{
+		Component:  getComponent(privateKeyName, privateKey, componentNameIdentifier, podinfoComponentName, version),
+		Repository: podinfoName,
 		ComponentVersionModifications: []shared.ComponentModification{
 			shared.ComponentVersionRef(shared.ComponentRef{
-				Name:          "backend",
-				Version:       "1.0.0",
+				Name:          backend,
+				Version:       version,
 				ComponentName: componentNamePrefix + componentNameIdentifier + podinfoBackendComponentName,
 			}),
 			shared.ComponentVersionRef(shared.ComponentRef{
-				Name:          "frontend",
-				Version:       "1.0.0",
+				Name:          frontend,
+				Version:       version,
 				ComponentName: componentNamePrefix + componentNameIdentifier + podinfoFrontendComponentName,
 			}),
 			shared.ComponentVersionRef(shared.ComponentRef{
-				Name:          "redis",
-				Version:       "1.0.0",
+				Name:          redis,
+				Version:       version,
 				ComponentName: componentNamePrefix + componentNameIdentifier + redisComponentName,
 			}),
 		},
 	}
-	return temp
-}
-
-func basicSignedComponent(t *testing.T, privateKey []byte, privateKeyName string, componentNameIdentifier string) setup.Component {
-	t.Helper()
-	temp := setup.Component{
-		Component: shared.Component{
-			Name:    componentNamePrefix + componentNameIdentifier + podinfoComponentName,
-			Version: "1.0.0",
-			Sign: &shared.Sign{
-				Name: privateKeyName,
-				Key:  privateKey,
-			},
-		},
-		Repository: "podinfo",
-		ComponentVersionModifications: []shared.ComponentModification{
-			shared.BlobResource(shared.Resource{
-				Name: "product-description",
-				Data: "test-component",
-				Type: "PlainText",
-			}),
-		},
-	}
-	return temp
 }
 
-func podinfoBackend(t *testing.T, componentNameIdentifier string) setup.Component {
+func podinfoBackend(t *testing.T, privateKey []byte, privateKeyName string, componentNameIdentifier string, testPath string, version string) setup.Component {
 	t.Helper()
 
-	configContent, err := os.ReadFile(filepath.Join(basePath, "podinfo", "backend", "config.yaml"))
+	configContent, err := os.ReadFile(filepath.Join(basePath, testPath, podinfoName, backend, "config.yaml"))
 	if err != nil {
-		t.Fatal("failed to read config file: %w", err)
+		t.Fatal(errMsgReadFile, err)
```

Comment:
No generalization here please.

Reply:
The error message tells you which file it was trying to add and failed. do we still need customer messages for each file?

Reply:
It's easier to search for when you are looking for an exact line or a specific message. Although the line number would most likely be included in the output, it's just a lot more convenient. :) That's why I don't like constructed error messages. They make it impossible to figure out where the error is coming from.